crates/nu-parser/src/parse_keywords.rs

```diff
@@ -3498,3 +3543,119 @@ pub fn find_in_dirs(
         }
     }
 }
+
+fn value_as_string(value: Value, span: Span) -> Result<String, ParseError> {
+    match value {
+        Value::String { val, .. } => Ok(val),
+        _ => Err(ParseError::NotAConstant(span)),
+    }
+}
+
+// similar to eval_expression() in the engine
+fn eval_constant(working_set: &StateWorkingSet, expr: &Expression) -> Result<Value, ParseError> {
+    match &expr.expr {
+        Expr::Bool(b) => Ok(Value::boolean(*b, expr.span)),
+        Expr::Int(i) => Ok(Value::int(*i, expr.span)),
+        Expr::Float(f) => Ok(Value::float(*f, expr.span)),
+        Expr::Binary(b) => Ok(Value::Binary {
+            val: b.clone(),
+            span: expr.span,
+        }),
+        Expr::Var(var_id) => match working_set.find_constant(*var_id) {
+            Some(val) => Ok(val.clone()),
+            None => Err(ParseError::NotAConstant(expr.span)),
+        },
+        Expr::FullCellPath(cell_path) => {
+            let value = eval_constant(working_set, &cell_path.head)?;
+
+            match value.follow_cell_path(&cell_path.tail, false) {
+                Ok(val) => Ok(val),
+                // TODO: Better error conversion
+                Err(shell_error) => Err(ParseError::LabeledError(
+                    "Error following cell path".to_string(),
+                    format!("{:?}", shell_error),
+                    expr.span,
+                )),
+            }
+        }
+        Expr::CellPath(cell_path) => Ok(Value::CellPath {
+            val: cell_path.clone(),
+            span: expr.span,
+        }),
+        Expr::DateTime(dt) => Ok(Value::Date {
+            val: *dt,
+            span: expr.span,
+        }),
+        Expr::Block(block_id) => Ok(Value::Block {
```

Comment:
For parsing performance, What about keeping `Expr::Block`, `Expr::List`, `Expr::Record`, `Expr::Table` out of our constant for now?

Like rust doesn't support making const variable which required heap allocation directly.

Reply:
I'm often using static arrays in C and Rust so I thought it would be useful to allow collections. You're probably unlikely to type massive slow collections by hand. But you're 100% right that we need to be super careful about what to allow so we don't end up with some sort of infinite loop or giant slow data structures.

Block seems safe to me but does it make sense having it as an constant? I don't know.

We'll need to experiment with these and choose carefully. Maybe start with a smaller subset and grow it over time. This is just a random selection I tried, I didn't even go through all the expressions.

Reply:
> Block seems safe to me but does it make sense having it as an constant? I don't know.

I think it's safe, but I don't think we need to make it as a constant.

> Maybe start with a smaller subset and grow it over time. 

Yeah, It's better than start with a larger one and shink it, that would be breaking change